### PR TITLE
Add platformio

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@ivankravets do you have interest in maintaining this package with me?

**Comment by hmaarrfk** [comment]
Currently this is failing on Mac due to 

https://github.com/conda-forge/pyelftools-feedstock/pull/21#issuecomment-565730562

**Comment by ivankravets** [comment]
@hmaarrfk Thanks for the PR! I do not use conda but I would be glad to help if you have any questions.

**Comment by hmaarrfk** [comment]
Got it, well I'll just remember to keep you in the loop in case I have any questions. No need for you to be a maintainer here if you don't want to. Since it is a pure python package, it is actually really easy to get this going.

**Comment by hmaarrfk** [comment]
friendly bump @conda-forge/staged-recipes

File: recipes/platformio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+test:
+  imports:
+    - platformio
+
+about:
+  home: https://platformio.org/
+  license: Apache-2.0
+  license_family: Apache
+  # Can remove versioned LICENSE file when:
+  # https://github.com/platformio/platformio-core/pull/3305
```
Comment (synapticarbors) [inline recipes/platformio/meta.yaml]: It looks like https://github.com/platformio/platformio-core/pull/3325 is merged

File: recipes/platformio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{% set name = "platformio" %}
+{% set version = "4.1.0" %}
```
Comment (synapticarbors) [inline recipes/platformio/meta.yaml]: Newest version is `4.2.1`

File: recipes/platformio/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 310fa8b624cf938ef95284ae3a40a9caa36aafef32ac30ad588b43df516945c3
+
+build:
+  Noarch: python
```
Comment (synapticarbors) [inline recipes/platformio/meta.yaml]: `Noarch` => 'noarch`

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**